src-java/flowmonitoring-topology/flowmonitoring-messaging/src/main/java/org/openkilda/messaging/info/flow/UpdateFlowInfo.java

```diff
@@ -65,4 +68,9 @@ public Long getMaxLatency() {
     public Long getMaxLatencyTier2() {
         return maxLatencyTier2 == null || maxLatencyTier2 == 0 ? Long.MAX_VALUE : maxLatencyTier2;
     }
+
+    public boolean isRemoveFlowFlowMonitoring() {
```

Comment:
Terrible name. `isRemoveRequest` fit better, but still not perfect.

PS If it is some sort of command, why it is `InfoMessage`? Looks like it should be `CommandMessage`.

Reply:
Reworked method name and `InfoMessage` to `CommandMessage`.

Actual bugfix was change `path.isEmpty()` -> `path.size() <= 1`. One switch flows will have path with one segment in this representation. And we need to remove such flows from monitoring.